### PR TITLE
[Fix bug 1120661] Respect group membership status in API results.

### PR DESCRIPTION
**Pull request by johngian**

### CODE REVIEW

**Comment by johngian** [comment]
@akatsoulas r?

File: mozillians/users/tests/test_api.py [hunk 1]
```diff
@@ -210,6 +210,21 @@ def test_search_groups(self):
...
         eq_(data['objects'][0]['id'], user_1.userprofile.id)
 
+    def test_search_groups_pending_membership(self):
+        client = Client()
+        group_1 = GroupFactory.create()
+        group_2 = GroupFactory.create()
+        user_1 = UserFactory.create()
+        user_2 = UserFactory.create()
+        group_1.add_member(user_1.userprofile)
+        group_2.add_member(user_2.userprofile, 'pending')
```
Comment (akatsoulas) [inline mozillians/users/tests/test_api.py]: s/pending/GroupMembership.PENDING/

Comment (akatsoulas) [inline mozillians/users/tests/test_api.py]: You don't need `group_2` for this patch. You need to add the user with the `Pending` state to the first group and then check that the user is not listed.

**Comment by johngian** [comment]
PR updated

File: mozillians/users/tests/test_api.py [hunk 2]
```diff
@@ -210,6 +211,21 @@ def test_search_groups(self):
         eq_(len(data['objects']), 1)
         eq_(data['objects'][0]['id'], user_1.userprofile.id)
 
+    def test_search_groups_pending_membership(self):
+        client = Client()
+        group_1 = GroupFactory.create()
+        group_2 = GroupFactory.create()
```
Comment (akatsoulas) [inline mozillians/users/tests/test_api.py]: Do you need this?

**Comment by akatsoulas** [comment]
lgtm r+

**Pull request merged**